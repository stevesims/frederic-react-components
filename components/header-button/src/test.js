import React from 'react';
import { mount, shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';
import { LINK_COLOUR } from 'govuk-colours';

import HeaderButton from '.';

expect.extend(createMatchers(emotion));

describe('HeaderButton', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<HeaderButton />);
    expect(wrapper.exists()).toBe(true);
  });

  it('applies styles on active prop', () => {
    const wrapper = shallow(<HeaderButton active />).find('Button');
    expect(wrapper).toHaveStyleRule('border-bottom-color', LINK_COLOUR);
    expect(wrapper).toHaveStyleRule('border-top-color', LINK_COLOUR);
    expect(wrapper).toHaveStyleRule('color', LINK_COLOUR);
  });

  it('applies styles on disabled prop', () => {
    const wrapper = shallow(<HeaderButton disabled />).find('Button');
    expect(wrapper).toHaveStyleRule('opacity', '0.35');
    // Can't see a way of testing pseudo-selectors here...
  });

  it('matches snapshot', () => {
    const wrapper = mount(<HeaderButton active />);
    expect(wrapper).toMatchSnapshot();
  });
});
