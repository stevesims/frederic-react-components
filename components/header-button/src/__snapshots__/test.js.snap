// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton matches snapshot 1`] = `
.emotion-0 {
  background: none;
  border-bottom: 4px solid transparent;
  border-left: none;
  border-right: none;
  border-top: 4px solid transparent;
  display: inline-block;
  font-family: "nta",Arial,sans-serif;
  font-size: 18px;
  font-weight: bold;
  min-width: 190px;
  padding: 8px;
  text-align: center;
  border-bottom-color: #005ea5;
  border-top-color: #005ea5;
  color: #005ea5;
}

.emotion-0:focus {
  background: #ffbf47;
  outline: 1px solid #ffbf47;
}

.emotion-0:hover {
  border-bottom-color: #2b8cc4;
  border-top-color: #2b8cc4;
}

<HeaderButton
  active={true}
>
  <Button
    active={true}
  >
    <button
      className="emotion-0 emotion-1"
    />
  </Button>
</HeaderButton>
`;
